### PR TITLE
Add alias for blockmean's -S parameter

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

According to #1598 this PR add an alias for the -S parameter of blockmean.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/blockm.py [hunk 1]
```diff
@@ -111,6 +112,12 @@ def blockmean(data=None, x=None, y=None, z=None, outfile=None, **kwargs):
 
     {I}
 
+    restype: str
+        Result type calculated by blockmean. Use **n** to report the number of
```
Comment (willschlitzer) [inline pygmt/src/blockm.py:116]:
```suggestion
        [**m**\|\ **n**\|\ **s**\|\ **w**].
        Result type calculated by blockmean. Use **n** to report the number of
```

I think this should have the potential arguments for `restype`

**Comment by willschlitzer** [comment]
@michaelgrund Sorry to leave you hanging, but I haven't had the chance to check out this branch yet!

**Comment by michaelgrund** [comment]
> 
> 
> @michaelgrund Sorry to leave you hanging, but I haven't had the chance to check out this branch yet!

No worries @willschlitzer 😉

**Comment by willschlitzer** [comment]
I'm not familiar with the use of `blockmean`, but I'm getting the exact same output when I change the arguments to `restype`. I'm using the input data specified in `test_blockmean_input_xyz()`. Is this supposed to be the case?

**Comment by meghanrjones** [comment]
> I'm not familiar with the use of `blockmean`, but I'm getting the exact same output when I change the arguments to `restype`. I'm using the input data specified in `test_blockmean_input_xyz()`. Is this supposed to be the case?

I get different results depending on the argument. Maybe try a git status to check the branch and `make clean; make install`?

```
import pygmt
data = pygmt.datasets.load_sample_bathymetry()
region = pygmt.info(data, spacing=1)
pygmt.blockmean(data=data, spacing="1m", restype="w", region=region).head(1)
```
longitude | latitude | bathymetry
-- | -- | --
245.88819 | 29.97895 | 1

```
pygmt.blockmean(data=data, spacing="1m", restype="m", region=region).head(1)
```
longitude | latitude | bathymetry
-- | -- | --
245.88819 | 29.97895 | -385.0

``restype`` is not my favorite alias for this parameter because I think truncated words are not the quickest to interpret. I would prefer something like ``statistic``. But, it's not that strong of an opinion if @michaelgrund prefers to stick with the ``restype``.

**Comment by michaelgrund** [comment]
> 
> `restype` is not my favorite alias for this parameter because I think truncated words are not the quickest to interpret. I would prefer something like `statistic`. But, it's not that strong of an opinion if @michaelgrund prefers to stick with the `restype`.

Thanks for you comment @meghanrjones. I'm fully open for other suggestions for the alias. Indeed, `statistic` would fit better than `restype`. Another alternative would be e.g. `summarize` since the calculated values represent a summary of the data in each block based on different metrics (mean, count, sum, sum of weights).

**Comment by seisman** [comment]
> > `restype` is not my favorite alias for this parameter because I think truncated words are not the quickest to interpret. I would prefer something like `statistic`. But, it's not that strong of an opinion if @michaelgrund prefers to stick with the `restype`.
> 
> Thanks for you comment @meghanrjones. I'm fully open for other suggestions for the alias. Indeed, `statistic` would fit better than `restype`. Another alternative would be e.g. `summarize` since the calculated values represent a summary of the data in each block based on different metrics (mean, count, sum, sum of weights).

Do you think `summary` (noun) is better than `summarize` (verb)?

**Comment by willschlitzer** [comment]
> > I'm not familiar with the use of `blockmean`, but I'm getting the exact same output when I change the arguments to `restype`. I'm using the input data specified in `test_blockmean_input_xyz()`. Is this supposed to be the case?
> 
> I get different results depending on the argument. Maybe try a git status to check the branch and `make clean; make install`?

Maybe I set up my Jupyter Notebook badly, but I'm now getting different results. Looks like it works!

**Comment by michaelgrund** [comment]
> Do you think `summary` (noun) is better than `summarize` (verb)?

I'm open for this, what do the others think is the better choice?

**Comment by willschlitzer** [comment]
> > Do you think `summary` (noun) is better than `summarize` (verb)?
> 
> I'm open for this, what do the others think is the better choice?

I think `summary` works better; `summarize` seems like it would make sense for a True/False argument.

File: pygmt/src/blockm.py [hunk 2]
```diff
@@ -111,6 +112,13 @@ def blockmean(data=None, x=None, y=None, z=None, outfile=None, **kwargs):
 
     {I}
 
+    summarize: str
+        [**m**\|\ **n**\|\ **s**\|\ **w**].
+        Type of summary values calculated by blockmean. Use **n** to report the
+        number of input points inside each block, **s** to report the sum of
+        all z-values inside a block, **w** to report the sum of weights
+        [Default or **m** reports mean value].
```
Comment (seisman) [inline pygmt/src/blockm.py:120]: List is more readable.
```suggestion
        Type of summary values calculated by blockmean. 
        
        - **n**: report the number of input points inside each block
        - **s**: report the sum of all z-values inside a block
        - **w**: report the sum of weights
        - **m**: report mean value [Default]
```

File: pygmt/src/blockm.py [hunk 3]
```diff
@@ -70,6 +70,7 @@ def _blockm(block_method, data, x, y, z, outfile, **kwargs):
 @use_alias(
     I="spacing",
     R="region",
+    S="summarize",
```
Comment (seisman) [inline pygmt/src/blockm.py:73]: As suggested in https://github.com/GenericMappingTools/pygmt/pull/1601#issuecomment-962215838:
```suggestion
    S="summary",
```

File: pygmt/src/blockm.py [hunk 4]
```diff
@@ -111,6 +112,13 @@ def blockmean(data=None, x=None, y=None, z=None, outfile=None, **kwargs):
 
     {I}
 
+    summarize: str
```
Comment (seisman) [inline pygmt/src/blockm.py:115]:
```suggestion
    summary: str
```

File: pygmt/src/blockm.py [hunk 5]
```diff
@@ -111,6 +112,15 @@ def blockmean(data=None, x=None, y=None, z=None, outfile=None, **kwargs):
 
     {I}
 
+    summary: str
+        [**m**\|\ **n**\|\ **s**\|\ **w**].
+        Type of summary values calculated by blockmean. 
+
```
Comment (seisman) [inline pygmt/src/blockm.py:118]: Need to remove these whitespaces to let the style checks pass.
```suggestion

```

**Review by seisman — APPROVED** [review]

File: pygmt/src/blockm.py (hunk 5 above)
Comment (weiji14) [reply pygmt/src/blockm.py]: Actually, you need to remove the whitespace, but keep the newline :sweat_smile:

File: pygmt/src/blockm.py [hunk 6]
```diff
@@ -111,6 +112,14 @@ def blockmean(data=None, x=None, y=None, z=None, outfile=None, **kwargs):
 
     {I}
 
+    summary: str
+        [**m**\|\ **n**\|\ **s**\|\ **w**].
+        Type of summary values calculated by blockmean.
+        - **n**: report the number of input points inside each block
+        - **s**: report the sum of all z-values inside a block
+        - **w**: report the sum of weights
+        - **m**: report mean value [Default]
```
Comment (seisman) [inline pygmt/src/blockm.py:121]: Now it looks like this:

image

I believe a new line is still needed at the start of the bullet.

**Review by seisman — APPROVED** [review]

**Review by willschlitzer — APPROVED** [review]

File: pygmt/src/blockm.py [hunk 7]
```diff
@@ -111,6 +112,15 @@ def blockmean(data=None, x=None, y=None, z=None, outfile=None, **kwargs):
...
     {I}
 
+    summary: str
+        [**m**\|\ **n**\|\ **s**\|\ **w**].
+        Type of summary values calculated by blockmean.
+
+        - **n**: report the number of input points inside each block
+        - **s**: report the sum of all z-values inside a block
+        - **w**: report the sum of weights
+        - **m**: report mean value [Default]
```
[the remark this answers is not in the document]
Comment (michaelgrund) [reply pygmt/src/blockm.py:122]: Good point!

**Pull request merged**